### PR TITLE
add authorized apps table to dashboard view

### PR DESCRIPTION
**Pull request by jwpincus**

### CODE REVIEW

File: app/controllers/apps_controller.rb [hunk 1]
```diff
@@ -40,6 +38,13 @@ def destroy
   end
   private
 
+  def app_owner?
+    if !App.find(params[:id]).owners.include?(current_user)
```
Comment (case-eee) [inline app/controllers/apps_controller.rb:42]: could you extract this logic out of the controller somehow? this is a lot for the controller to know about!

File: spec/features/user_can_see_authorized_apps_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+describe 'A user can' do
+  before(:each) do
+    user = create(:enrolled_user)
+    @app_user = create(:enrolled_user)
+    @app = create(:app)
+    user.apps << @app
+    @app.authorized_users << @app_user
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@app_user)
```
Comment (case-eee) [inline spec/features/user_can_see_authorized_apps_spec.rb:12]: 👍

**Review by case-eee — COMMENTED** [review]
good work @jwpincus 👍

File: app/controllers/apps_controller.rb (hunk 1 above)
Comment (jwpincus) [reply app/controllers/apps_controller.rb:42]: @case-eee Hmm. This is an authorization control before_action method to keep unauthorized users away from show, update, delete actions for apps they don't own. Is this something that can be controlled farther down the stack? I can push it to ApplicationController if need be, but not sure how to get it out of a controller altogether/

Comment (case-eee) [reply app/controllers/apps_controller.rb:42]: @jwpincus you can't extract the entire method itself, but we could extract the logic within the method. you'd need to write a method that takes in the id and user like so...

```
class App
  def self.check_owner?(id, user)
    find(id).owners.include?(user)
  end
end
```

then, in your method in your controller, you could have..

```
if !App.check_owner?(params[:id], current_user)
```

**Pull request merged**